db/class.go

```diff
@@ -178,56 +178,45 @@ func (d *DB) CreateClass(c echo.Context) error {
 
 // GetClass takes the UID (either of a member or an instructor)
 // and a CID (wid) as a JSON, and returns an object representing the class.
-// If the given UID is not a member or an instructor, error is returned
+// If the given UID is not a member or an instructor, an error is returned.
 func (d *DB) GetClass(c echo.Context) error {
-	var req struct {
-		UID string `json:"uid"`
-		WID string `json:"wid"`
-	}
+	var (
+		req struct {
+			UID string `json:"uid"`
+			CID string `json:"cid"`
+		}
+		res struct {
+			*Class
+			ProgramData []Program       `json:"programData"`
+			UserData    map[string]User `json:"userData"`
+		}
+		err error
+	)
+
 	if err := httpext.RequestBodyTo(c.Request(), &req); err != nil {
 		return c.String(http.StatusInternalServerError, errors.Wrap(err, "failed to read request body").Error())
 	}
-	if req.UID == "" || req.WID == "" {
+	if req.UID == "" || req.CID == "" {
 		return c.String(http.StatusBadRequest, "uid and wid fields are both required")
```

Comment:
Think this could be `cid` instead of `wid`?
```suggestion
		return c.String(http.StatusBadRequest, "uid and cid fields are both required")
```

Reply:
Nice catch!